draft-ietf-httpbis-cache-latest.xml

```diff
@@ -558,6 +563,13 @@
    response (as determined by the <x:ref>Date</x:ref> header field) is used, as
    per <xref target="constructing.responses.from.caches"/>.
 </t>
+<t>
+   Note that in practice, some resources might omit appropriate Vary header
+   values on some representations, or otherwise have Vary headers whose values
+   aren't stable. Caches &MAY; apply heuristics to identify such resources and
+   act appropriately (typically, applying the most recent Vary header value to
+   all responses that share the same target URI).
+</t>
```

Comment:
FWIW, isn't that sort-of required? It would be the result of the server changing Vary dimensions, in which case the latest would definitvely the right one to consider.

Reply:
Yeah, I don't see the point of adding this. In general, the origin server chooses what dimensions of variance it wants a cache to care about, and anything not mentioned is assumed to be "okay". The intent is that the most recent acceptable fresh response will be reused by the cache, followed by most recent acceptable stale response. Further heuristics are always possible, but that isn't an interop requirement.

Reply:
Many many servers are incorrectly implemented, and oscillate between different vary values, e.g., omitting them on the "default" representation. If a cache only pays attention to the latest or the values on the cached responses in isolation, it will behave incorrectly. Yes, this is the server's fault, but it seems reasonable to warn implementers about this widely experienced condition.

Reply:
Text revised, PTAL